test/Exercise/DependencyHeavenTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+
+namespace PhpSchool\LearnYouPhpTest\Exercise;
+
+use Faker\Factory;
+use Faker\Generator;
+use PhpSchool\LearnYouPhp\Exercise\DependencyHeaven;
+use PhpSchool\PhpWorkshop\Result\Failure;
+use PhpSchool\PhpWorkshop\Result\Success;
+use PHPUnit_Framework_TestCase;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Class DependencyHeavenTest
+ * @package PhpSchool\LearnYouPhpTest\Exercise
+ * @author Michael Woodward <mikeymike.mw@gmail.com>
+ */
+class DependencyHeavenTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Generator
+     */
+    private $faker;
+
+    public function setUp()
+    {
+        $this->faker = Factory::create('Fr_fr');
+    }
+
+    public function testDependencyHeavenExercise()
+    {
+        $e = new DependencyHeaven($this->faker);
+
+        $this->assertEquals('Dependency Heaven', $e->getName());
+        $this->assertEquals('An introduction to Composer dependency management', $e->getDescription());
+
+        $this->assertFileExists(realpath($e->getSolution()));
+        $this->assertFileExists(realpath($e->getProblem()));
+        $this->assertNull($e->tearDown());
+    }
+
+    public function testGetRequests()
+    {
+        $e = new DependencyHeaven($this->faker);
+
+        $requests  = $e->getRequests();
+
+        foreach ($requests as $request) {
+            $this->assertInstanceOf(RequestInterface::class, $request);
+            $this->assertSame('POST', $request->getMethod());
+            $this->assertSame(['application/x-www-form-urlencoded'], $request->getHeader('Content-Type'));
+            $this->assertNotEmpty($request->getBody());
+        }
+    }
+
+    public function testGetRequestsReturnsMultipleRequestsForEachEnpoint()
+    {
+        $e = new DependencyHeaven($this->faker);
```

Comment:
Doing that would only verify there is one occurrence, this is testing to ensure there are multiple for each endpoint. Plus the amount of requests per endpoint is random atm, although I could add this to the construct and test a specific amount of occurrences if you think it will add benefit.